src/components/common/Interactive.tsx

```diff
@@ -108,36 +101,26 @@ const InteractiveBase = ({ onMove, onKey, ...rest }: Props) => {
         left: keyCode === 39 ? 0.05 : keyCode === 37 ? -0.05 : 0,
         top: keyCode === 40 ? 0.05 : keyCode === 38 ? -0.05 : 0,
       });
-    },
-    [onKeyCallback]
-  );
-
-  const handleMoveEnd = useCallback(() => setDragging(false), []);
+    };
 
-  const toggleDocumentEvents = useCallback(
-    (state) => {
+    function toggleDocumentEvents(state: boolean) {
+      const touch = hasTouch.current;
       // add or remove additional pointer event listeners
-      const toggleEvent = state ? window.addEventListener : window.removeEventListener;
-      toggleEvent(hasTouched.current ? "touchmove" : "mousemove", handleMove);
-      toggleEvent(hasTouched.current ? "touchend" : "mouseup", handleMoveEnd);
-    },
-    [handleMove, handleMoveEnd]
-  );
+      const toggleEvent = state ? self.addEventListener : self.removeEventListener;
```

Comment:
self is supported by all the browsers and it's smaller than window

Reply:
Smart! 👌